src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -174,6 +175,163 @@ public static void ClientBaseOfT_Sync_RoundTrip_Call_Using_NetTcpTransport()
+
+        try
+        {
+            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+            IWcfService serviceProxy = factory.CreateChannel();
+
+            Stream stream = StringToStream(testString);
+            var returnStream = serviceProxy.EchoStream(stream);
+            var result = StreamToString(returnStream);
+            success = string.Equals(result, testString);
+
+            if (!success)
+            {
+                errorBuilder.AppendLine(String.Format("    Error: expected response from service: '{0}' Actual was: '{1}'", testString, result));
+            }
+        }
+        catch (Exception ex)
+        {
+            errorBuilder.AppendLine(String.Format("    Error: Unexpected exception was caught while doing the basic echo test for variation...\n'{0}'\nException: {1}", variationDetails, ex.ToString()));
+        }
+
+        Assert.True(errorBuilder.Length == 0, "Test case FAILED with errors: " + errorBuilder.ToString());
+    }
+
+    [Fact]
+    [OuterLoop]
+    public static void DefaultSettings_Echo_RoundTrips_String_Streamed_Async()
+    {
+        string variationDetails = "Client:: BasicHttpBinding/DefaultValues\nServer:: BasicHttpBinding/DefaultValues";
+        string testString = "Hello";
+        StringBuilder errorBuilder = new StringBuilder();
+        bool success = false;
+
+        BasicHttpBinding binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+        binding.TransferMode = TransferMode.Streamed;
+
+        try
+        {
+            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+            IWcfService serviceProxy = factory.CreateChannel();
+
+            Stream stream = StringToStream(testString);
+            var returnStream = serviceProxy.EchoStreamAsync(stream).Result;
+            var result = StreamToString(returnStream);
+            success = string.Equals(result, testString);
+
+            if (!success)
+            {
+                errorBuilder.AppendLine(String.Format("    Error: expected response from service: '{0}' Actual was: '{1}'", testString, result));
+            }
+        }
+        catch (Exception ex)
+        {
+            errorBuilder.AppendLine(String.Format("    Error: Unexpected exception was caught while doing the basic echo test for variation...\n'{0}'\nException: {1}", variationDetails, ex.ToString()));
+        }
+
+        Assert.True(errorBuilder.Length == 0, "Test case FAILED with errors: " + errorBuilder.ToString());
+    }
+
+    private static string StreamToString(Stream stream)
+    {
+        var reader = new StreamReader(stream, Encoding.Unicode);
+        return reader.ReadToEnd();
+    }
+
+    private static Stream StringToStream(string str)
+    {
+        var ms = new MemoryStream();
+        var sw = new StreamWriter(ms, Encoding.Unicode);
```

Comment:
UTF8 is what I've generally used in the past and what's out in the wild a lot; I'm not too concerned about it either way though.